api/sandbox.go

```diff
@@ -0,0 +1,114 @@
+package api
+
+import (
+	"context"
+	"syscall"
+
+	"github.com/hyperhq/runv/api"
+	ocispecs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+type ContainerConfig struct {
+	Id string
+
+	UGI    *UserGroupInfo
+	Mounts map[string]Mount
+
+	OciSpec ocispecs.Spec
+}
+
+type Mount struct {
+	StorageID string
+	Path      string
+	ReadOnly  bool
+}
+
+type UserGroupInfo struct {
+	User             string
+	Group            string
+	AdditionalGroups []string
+}
+
+type ProcessConfig struct {
+	Container string
+	Id        string
+	UGI       *UserGroupInfo
+
+	OciProcess ocispecs.Process
+}
+
```

Comment:
I think the virtcontainers equivalent for this one is `Cmd`:

```Go
// Cmd represents a command to execute in a running container.
type Cmd struct {
	Args    []string
	Envs    []EnvVar
	WorkDir string

	// Note that these fields *MUST* remain as strings.
	//
	// The reason being that we want runtimes to be able to support CLI
	// operations like "exec --user=". That option allows the
	// specification of a user (either as a string username or a numeric
	// UID), and may optionally also include a group (groupame or GID).
	//
	// Since this type is the interface to allow the runtime to specify
	// the user and group the workload can run as, these user and group
	// fields cannot be encoded as integer values since that would imply
	// the runtime itself would need to perform a UID/GID lookup on the
	// user-specified username/groupname. But that isn't practically
	// possible given that to do so would require the runtime to access
	// the image to allow it to interrogate the appropriate databases to
	// convert the username/groupnames to UID/GID values.
	//
	// Note that this argument applies solely to the _runtime_ supporting
	// a "--user=" option when running in a "standalone mode" - there is
	// no issue when the runtime is called by a container manager since
	// all the user and group mapping is handled by the container manager
	// and specified to the runtime in terms of UID/GID's in the
	// configuration file generated by the container manager.
	User                string
	PrimaryGroup        string
	SupplementaryGroups []string

	Interactive     bool
	Console         string
	Detach          bool
	NoNewPrivileges bool
	Capabilities    LinuxCapabilities
}
```